src/tooltip-manager.ts

```diff
@@ -0,0 +1,137 @@
+'use babel';
+
+/// <reference path="marked/index.d.ts" />
+
+import {CompositeDisposable} from 'atom';
+import {RemoteEditorService} from 'polymer-analyzer/lib/editor-service/remote-editor-service';
+import * as marked from 'marked';
+
+class TooltipManager {
+  textEditor: AtomCore.IEditor;
+  tooltipMarker: AtomCore.IDisplayBufferMarker;
+  tooltipElement: HTMLElement = null;
+  documentationElement: HTMLElement = null;
+  editorService: RemoteEditorService;
+  editorElements: Array<HTMLElement> = [];
+  cursorOutsideTooltip: boolean = true;
+
+  constructor(editorService: RemoteEditorService) {
+    this.editorService = editorService;
+  };
+
+  subscribe(subscriptions: CompositeDisposable) {
+    subscriptions.add(atom.workspace.observePanes((pane: AtomCore.IPane) => {
+      this.update();
+      subscriptions.add(pane.onDidChangeActiveItem(() => {
+        this.update();
+      }));
+    }));
+  };
+
+  update() {
+    this.removeTooltip();
+    this.textEditor = atom.workspace.getActiveTextEditor();
+    this.addMouseEventListener();
+  }
+
+  addMouseEventListener() {
+    const textEditorElement = atom.views.getView(this.textEditor);
+    if (!textEditorElement) {
+      return;
+    }
+    if (this.editorElements.indexOf(textEditorElement) > -1) {
+      return;
+    }
+    this.editorElements.push(textEditorElement);
+    let timer: number;
+    textEditorElement.addEventListener('mousemove', (event: MouseEvent) => {
+      window.clearTimeout(timer);
+      if (this.cursorOutsideTooltip) {
+        this.removeTooltip();
+        timer = window.setTimeout(() => this.calculatePositions(event, textEditorElement), 100);
+      }
+    });
+  };
+
+  calculatePositions(event: MouseEvent, textEditorElement: any) {
+    // The text editor could have been closed before the timeout was fired
+    if (!textEditorElement || !textEditorElement.component) {
+      return;
+    }
+    const reportedCursorPosition = textEditorElement.component.screenPositionForMouseEvent(event);
+    const currentCursorPixelPosition = textEditorElement.component.pixelPositionForMouseEvent(event);
+    const expectedPixelPosition = textEditorElement.pixelPositionForScreenPosition(reportedCursorPosition);
+
+    const nearColumn = Math.abs(currentCursorPixelPosition.left - expectedPixelPosition.left) < 20;
+    const nearRow = Math.abs(currentCursorPixelPosition.top - expectedPixelPosition.top) < 20;
+
+    if (nearColumn && nearRow) {
+      this.updateTooltip(this.textEditor.bufferPositionForScreenPosition(reportedCursorPosition));
+    }
+  }
+
+  async updateTooltip(point: {row: number, column: number}) {
+    this.removeTooltip();
+
+    if (!(this.editorService && this.textEditor)) {
+      return;
+    }
+    const relativePath: string =
+      atom.project['relativizePath'](this.textEditor.getPath())[1];
```

Comment:
Ah this one is also defined in `_lint` and `_getSuggestions`, I copied from there. Should I change these too?